packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
+import { Primitive } from '@radix-ui/react-primitive';
+import { getSelector } from '@radix-ui/utils';
+import { createContext, composeEventHandlers } from '@radix-ui/react-utils';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+import type { Merge } from '@radix-ui/utils';
+import { Separator as SeparatorImpl } from '@radix-ui/react-separator';
+
+/* -------------------------------------------------------------------------------------------------
+ * Toolbar
+ * -----------------------------------------------------------------------------------------------*/
+
+const TOOLBAR_NAME = 'Toolbar';
+const TOOLBAR_CONTEXT_NAME = TOOLBAR_NAME + 'Context';
+const TOOLBAR_DEFAULT_TAG = 'div';
+
+type Orientation = React.AriaAttributes['aria-orientation'];
+
+type RovingFocusGroupProps = React.ComponentProps<typeof RovingFocusGroup>;
+
+type ToolbarOwnProps = Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    /**
+     * The orientation of the toolbar.
+     * Mainly so arrow navigation is done accordingly (left & right vs. up & down)
+     * (default: horizontal)
+     */
+    orientation?: RovingFocusGroupProps['orientation'];
+    /**
+     * The direction of navigation between toolbar items.
+     * (default: ltr)
+     */
+    dir?: RovingFocusGroupProps['dir'];
+    /**
+     * Whether keyboard navigation should loop focus
+     * (default: true)
+     */
+    loop?: RovingFocusGroupProps['loop'];
+  }
+>;
+
+type ToolbarPrimitive = Polymorphic.ForwardRefComponent<
+  typeof TOOLBAR_DEFAULT_TAG,
```

Comment:
Because it's using `Primitive` under the hood and we're no longer passing `as={Primitive}` we can make sure this uses the right type from `Primitive` by doing the following:

```jsx
type ToolbarPrimitive = Polymorphic.ForwardRefComponent<
  Polymorphic.IntrinsicElement<typeof Primitive>,
```

And then we can remove the `TOOLBAR_DEFAULT_TAG` const here: https://github.com/radix-ui/primitives/pull/412/files#diff-2154e6bce0230d6ec3e2261a22c57f1dc5167011fb45b3c4da1921f356e91bedR16